sympy/ntheory/factor_.py

```diff
@@ -1309,17 +1309,17 @@ def factorint(n, limit=None, use_trial=True, use_rho=True, use_pm1=True,
             # square root anyway. Finding 2 factors is easy if they are
             # "close enough." This is the big root equivalent of dividing by
             # 2, 3, 5.
-            sqrt_n = integer_nthroot(n, 2)[0]
+            sqrt_n = sqrt(n)
```

Comment:
`is `sqrt(n)` going to be an integer?

Reply:
I apologize for the confusion.
In addition to moving `is_square`, this PR also does the following:

* Replace `gmpy.sqrt` where it can be used.
  * Change `integer_nthroot(n, 2)[0]` to `sqrt(n)`.
* Replace `gmpy.sqrtrem` where it can be used.
  * Change `integer_nthroot(n, 2)` to `sqrtrem(n)`.